src/near_sdk_py/collections/unordered_map.py

```diff
@@ -84,17 +84,19 @@ def __iter__(self) -> Iterator:
         """Return an iterator over the keys"""
         return iter(self._keys_vector)
 
-    def keys(self) -> List:
-        """Return a list of all keys"""
-        return list(self._keys_vector)
+    def keys(self) -> Iterator[Any]:
+        """Return an iterator over the keys"""
+        return iter(self._keys_vector)
 
-    def values(self) -> List:
-        """Return a list of all values"""
-        return [self[key] for key in self._keys_vector]
+    def values(self) -> Iterator[Any]:
+        """Return an iterator over the values"""
+        for key in self._keys_vector:
+            yield self[key]
 
-    def items(self) -> List[Tuple]:
-        """Return a list of all (key, value) pairs"""
-        return [(key, self[key]) for key in self._keys_vector]
+    def items(self) -> Iterator[Tuple[Any, Any]]:
+        """Return an iterator over the (key, value) pairs"""
+        for key in self._keys_vector:
+            yield (key, self[key])
```

Comment:
Can I `.skip(1000).take(20)` and not run out of gas? If not, there is no point in this abstraction and it is as good as LookupMap. This used to be also a problem with Rust collections and this is why we "renamed" them into `IterableMap` to highlight the use-case and make sure the SDK devs also keep this use-case in mind.